packages/multichain-account-service/src/providers/SolAccountProvider.ts

```diff
@@ -102,10 +124,13 @@ export class SolAccountProvider extends SnapAccountProvider {
     entropySource: EntropySourceId;
     groupIndex: number;
   }): Promise<Bip44Account<KeyringAccount>[]> {
-    const discoveredAccounts = await this.#client.discoverAccounts(
-      [SolScope.Mainnet],
-      entropySource,
-      groupIndex,
+    const discoveredAccounts = await withTimeout(
```

Comment:
Do we know that the solana snap does not have a timeout mechanism in place already for discovery? I'm also thinking that we should have the timeout & retry stuff mentioned in the architecture documentation as well just so snap providers aren't implementing this stuff within the snap itself. Albeit small, doing this on the service side actually reduces some code duplication (if they are doing some retry/time-out mechanism).

Reply:
I haven't check on the Solana Snap, but I 100% agree with you that we have to implement this at service level.

Mainly cause we don't want this to start failing if the Snap is buggy or anything like that.

Regarding the retry, we could also add it there too, yes, this way we would have a similar way (for each built-in providers) to run the discovery 👍

I'll add this!